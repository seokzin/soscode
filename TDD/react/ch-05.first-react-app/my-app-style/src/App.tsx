import React from 'react';
import logo from './logo.svg';
import Styled, {keyframes} from 'styled-components';

const Container = Styled.div`
  text-align: center;
`;

const Header = Styled.header`
  background-color: #282c34 !important;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const spin = keyframes`
  from {
      transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;


const AppLogo = Styled.img`
  height: 40vmin;
  pointer-events: none;

  @media (prefers-reduced-motion: no-preference) {
    animation: ${spin} infinite 20s linear;
  }
`;

const AppLink = Styled.a`
  color: #61dafb;
`;

function App() {
  return (
    // <div className="App">
    <Container>
      {/* <header className="App-header"> */}
      <Header>
        {/* <img src={logo} className="App-logo" alt="logo" /> */}
        <AppLogo src={logo} alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <AppLink
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </AppLink>
      </Header>
      {/* </header> */}
    </Container>
    // </div>
  );
}

export default App;
