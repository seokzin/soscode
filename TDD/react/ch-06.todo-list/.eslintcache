[{"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/index.tsx":"1","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/App.tsx":"3","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/index.tsx":"4","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Button/index.tsx":"5","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Input/index.tsx":"6","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/ToDoItem/index.tsx":"7"},{"size":501,"mtime":1608450377531,"results":"8","hashOfConfig":"9"},{"size":425,"mtime":1607145602318,"results":"10","hashOfConfig":"9"},{"size":1672,"mtime":1615884272454,"results":"11","hashOfConfig":"9"},{"size":79,"mtime":1608450376614,"results":"12","hashOfConfig":"9"},{"size":989,"mtime":1608450392527,"results":"13","hashOfConfig":"9"},{"size":644,"mtime":1610068351702,"results":"14","hashOfConfig":"9"},{"size":655,"mtime":1608450377481,"results":"15","hashOfConfig":"9"},{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},"1o6yy7c",{"filePath":"19","messages":"20","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"21","messages":"22","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"23","messages":"24","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"25","messages":"26","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"27","messages":"28","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"29"},{"filePath":"30","messages":"31","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/index.tsx",[],["32","33"],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/App.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Button/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Input/index.tsx",[],["34","35"],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/ToDoItem/index.tsx",[],{"ruleId":"36","replacedBy":"37"},{"ruleId":"38","replacedBy":"39"},{"ruleId":"36","replacedBy":"40"},{"ruleId":"38","replacedBy":"41"},"no-native-reassign",["42"],"no-negated-in-lhs",["43"],["42"],["43"],"no-global-assign","no-unsafe-negation"]