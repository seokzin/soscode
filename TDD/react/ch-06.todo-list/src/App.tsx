import React, { useState } from 'react';
import Styled from 'styled-components';

import { Button, Input, ToDoItem } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
const Contents = Styled.div`
  display: flex;
  background-color: #FFFFFF;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;
const ToDoListContainer = Styled.div`
  min-width: 350px;
  height: 400px;
  overflow-y: scroll;
  border: 1px solid #BDBDBD;
  margin-bottom: 20px;
`;
const InputContainer = Styled.div`
  display: flex;
`;

function App() {
  const [toDo, setToDo] = useState('');
  const [toDoList, setToDoList] = useState<string[]>([]);

  const addToDo = (): void => {
    if (toDo) {
      setToDoList([...toDoList, toDo]);
      setToDo('');
    }
  };

  const deleteToDo = (index: number): void => {
    let list = [...toDoList];
    list.splice(index, 1);
    setToDoList(list);
  };

  return (
    <Container>
      <Contents>
        <ToDoListContainer data-testid='toDoList'>
          {toDoList.map((item, index) => (
            <ToDoItem key={item} label={item} onDelete={() => deleteToDo(index)} />
          ))}
        </ToDoListContainer>
        <InputContainer>
          <Input
            placeholder="할 일을 입력해 주세요"
            value={toDo}
            onChange={(text) => setToDo(text)}
          />
          <Button label="추가" onClick={addToDo} />
        </InputContainer>
      </Contents>
    </Container>
  );
}

export default App;
