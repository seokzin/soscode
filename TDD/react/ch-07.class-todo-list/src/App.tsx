import React, { Component } from 'react';
import type { IScriptSnapshot } from 'typescript';
import Styled from 'styled-components';

import { Button, Input, ToDoItem } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
const Contents = Styled.div`
  display: flex;
  background-color: #FFFFFF;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;
const ToDoListContainer = Styled.div`
  min-width: 350px;
  height: 400px;
  overflow-y: scroll;
  border: 1px solid #BDBDBD;
  margin-bottom: 20px;
`;
const InputContainer = Styled.div`
  display: flex;
`;

interface Props {}

interface State {
  readonly toDo: string;
  readonly toDoList: string[];
  readonly error: boolean;
}

class App extends Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      toDo: '',
      toDoList: [],
      error: false,
    };
  }

  private addToDo = (): void => {
    const { toDo, toDoList } = this.state;
    if (toDo) {
      this.setState({
        toDo: '',
        toDoList: [...toDoList, toDo],
      });
    }
  };

  private deleteToDo = (index: number): void => {
    let list = [...this.state.toDoList];
    list.splice(index, 1);
    this.setState({
      toDoList: list,
    });
  };

  render() {
    const { toDo, toDoList, error } = this.state;

    return (
      <Container>
        {!error && (
          <Contents>
            <ToDoListContainer data-testid="toDoList">
              {toDoList.map((item, index) => (
                <ToDoItem key={item} label={item} onDelete={() => this.deleteToDo(index)} />
              ))}
            </ToDoListContainer>
            <InputContainer>
              <Input
                placeholder="할 일을 입력해 주세요"
                value={toDo}
                onChange={(text) => this.setState({ toDo: text })}
              />
              <Button label="추가" onClick={this.addToDo} />
            </InputContainer>
          </Contents>
        )}
      </Container>
    );
  }

  // static getDerivedStateFromProps(nextProps: Props, prevState: State) {
  //   console.log('getDerivedStateFromProps');

  //   return null;
  // }

  // componentDidMount() {
  //   console.log('componentDidMount');
  // }

  // getSnapshotBeforeUpdate(prevProps: Props, prevState: State) {
  //   console.log('getSnapshotBeforeUpdate');

  //   return {
  //     testData: true,
  //   };
  // }

  // componentDidUpdate(prevProps: Props, prevState: State, snapshot: IScriptSnapshot) {
  //   console.log('componentDidUpdate');
  // }

  // shouldComponentUpdate(nextProps: Props, nextState: State) {
  //   console.log('shouldComponentUpdate');
  //   return true;
  // }

  // componentWillUnmount() {
  //   console.log('componentWillUnmount');
  // }

  // componentDidCatch(error: Error, info: React.ErrorInfo) {
  //   this.setState({
  //     error: true,
  //   });
  // }
}
export default App;
