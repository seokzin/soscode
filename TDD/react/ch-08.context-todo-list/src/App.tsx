import React from 'react';
import Styled from 'styled-components';

import { ToDoListProvider } from 'Contexts';
import { InputContainer, ToDoList } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
const Contents = Styled.div`
  display: flex;
  background-color: #FFFFFF;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;

function App() {
  return (
    <ToDoListProvider>
      <Container>
        <Contents>
          <ToDoList />
          <InputContainer />
        </Contents>
      </Container>
    </ToDoListProvider>
  );
}

export default App;
