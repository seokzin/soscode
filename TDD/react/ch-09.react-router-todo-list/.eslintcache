[{"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/index.tsx":"1","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/ToDoList/index.tsx":"2","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Detail/index.tsx":"3","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Add/index.tsx":"4","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Contexts/ToDoList/index.tsx":"5","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/NotFound/index.tsx":"6","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/InputContainer/index.tsx":"7","/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/PageHeader/index.tsx":"8"},{"size":590,"mtime":1611741896766,"results":"9","hashOfConfig":"10"},{"size":634,"mtime":1610777121363,"results":"11","hashOfConfig":"10"},{"size":1109,"mtime":1611366664483,"results":"12","hashOfConfig":"10"},{"size":565,"mtime":1610775384887,"results":"13","hashOfConfig":"10"},{"size":1295,"mtime":1610774495642,"results":"14","hashOfConfig":"10"},{"size":204,"mtime":1610952135141,"results":"15","hashOfConfig":"10"},{"size":833,"mtime":1610955003283,"results":"16","hashOfConfig":"10"},{"size":1023,"mtime":1611743640430,"results":"17","hashOfConfig":"10"},{"filePath":"18","messages":"19","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"20"},"7xskaz",{"filePath":"21","messages":"22","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"23","messages":"24","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"25","messages":"26","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"27","messages":"28","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"29","messages":"30","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"31","messages":"32","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"33","messages":"34","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/index.tsx",[],["35","36"],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/ToDoList/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Detail/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/Add/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Contexts/ToDoList/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Pages/NotFound/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/InputContainer/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch9.react-router-todo-list/src/Components/PageHeader/index.tsx",[],{"ruleId":"37","replacedBy":"38"},{"ruleId":"39","replacedBy":"40"},"no-native-reassign",["41"],"no-negated-in-lhs",["42"],"no-global-assign","no-unsafe-negation"]