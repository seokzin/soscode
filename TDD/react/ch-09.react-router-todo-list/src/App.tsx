import React from 'react';
import Styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import { ToDoListProvider } from 'Contexts';
import { List, Add, Detail, NotFound } from 'Pages';
import { PageHeader } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

function App() {
  return (
    <ToDoListProvider>
      <Container>
        <PageHeader />
        <Switch>
          <Route exact path="/">
            <List />
          </Route>
          <Route path="/add">
            <Add />
          </Route>
          <Route path="/detail/:id">
            <Detail />
          </Route>
          <Route>
            <NotFound />
          </Route>
        </Switch>
      </Container>
    </ToDoListProvider>
  );
}

export default App;
