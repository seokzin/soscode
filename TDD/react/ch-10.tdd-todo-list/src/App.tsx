import React from 'react';
import Styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import { PageHeader } from 'Components';
import { List, Add, Detail, NotFound } from 'Pages';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

function App() {
  return (
    <Container>
      <PageHeader />
      <Switch>
        <Route exact path="/">
          <List />
        </Route>
        <Route path="/add">
          <Add />
        </Route>
        <Route path="/detail/:id">
          <Detail />
        </Route>
        <Route>
          <NotFound />
        </Route>
      </Switch>
    </Container>
  );
}

export default App;
