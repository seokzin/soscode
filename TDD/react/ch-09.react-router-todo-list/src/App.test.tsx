import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import 'jest-styled-components';

describe('<App />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/');

    const { container } = render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const header = screen.getByText('Ìï† Ïùº Î™©Î°ù');
    expect(header).toBeInTheDocument();

    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
    expect(toDoList.firstChild).toBeNull();

    const label = screen.getByText('+');
    expect(label).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('goes to Add page and go back to List page', () => {
    const history = createMemoryHistory();
    history.push('/');

    const { container } = render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const addButton = screen.getByText('+');
    fireEvent.click(addButton);

    const header = screen.getByText('Ìï† Ïùº Ï∂îÍ∞Ä');
    expect(header).toBeInTheDocument();
    const goBack = screen.getByText('ÎèåÏïÑÍ∞ÄÍ∏∞');
    expect(goBack).toBeInTheDocument();
    const input = screen.getByPlaceholderText('Ìï† ÏùºÏùÑ ÏûÖÎ†•Ìï¥ Ï£ºÏÑ∏Ïöî');
    expect(input).toBeInTheDocument();
    const button = screen.getByText('Ï∂îÍ∞Ä');
    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();

    fireEvent.click(goBack);
    expect(header.textContent).toBe('Ìï† Ïùº Î™©Î°ù');
    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
  });

  it('goes to Detail page and go back to List page', () => {
    localStorage.setItem('ToDoList', '["ToDo 1"]');

    const history = createMemoryHistory();
    history.push('/');

    const { container } = render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const toDoItem = screen.getByText('ToDo 1');
    expect(toDoItem).toBeInTheDocument();
    fireEvent.click(toDoItem);

    const header = screen.getByText('Ìï† Ïùº ÏÉÅÏÑ∏');
    expect(header).toBeInTheDocument();
    const goBack = screen.getByText('ÎèåÏïÑÍ∞ÄÍ∏∞');
    expect(goBack).toBeInTheDocument();
    const toDo = screen.getByText('ToDo 1');
    expect(toDo).toBeInTheDocument();
    const button = screen.getByText('ÏÇ≠Ï†ú');
    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();

    fireEvent.click(goBack);
    expect(header.textContent).toBe('Ìï† Ïùº Î™©Î°ù');
    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
  });

  it('shows Not Found page if the user enters the wrong URL, and go back to List page', () => {
    const history = createMemoryHistory();
    history.push('/foo');

    const { container } = render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const header = screen.getByText('ÏóêÎü¨');
    expect(header).toBeInTheDocument();
    const goBack = screen.getByText('ÎèåÏïÑÍ∞ÄÍ∏∞');
    expect(goBack).toBeInTheDocument();
    const notFoundMessage = screen.getByText('Not Found üòø');
    expect(notFoundMessage).toBeInTheDocument();

    expect(container).toMatchSnapshot();

    fireEvent.click(goBack);
    expect(header.textContent).toBe('Ìï† Ïùº Î™©Î°ù');
    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
  });

  it('adds a new ToDo', () => {
    const history = createMemoryHistory();
    history.push('/');

    render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const addButton = screen.getByText('+');
    fireEvent.click(addButton);

    const input = screen.getByPlaceholderText('Ìï† ÏùºÏùÑ ÏûÖÎ†•Ìï¥ Ï£ºÏÑ∏Ïöî');
    const button = screen.getByText('Ï∂îÍ∞Ä');
    fireEvent.change(input, { target: { value: 'New ToDo' } });
    fireEvent.click(button);

    const header = screen.getByText('Ìï† Ïùº Î™©Î°ù');
    expect(header).toBeInTheDocument();
    const newToDo = screen.getByText('New ToDo');
    expect(newToDo).toBeInTheDocument();
  });

  it('deletes ToDo from ToDo List page', () => {
    localStorage.setItem('ToDoList', '["ToDo 1"]');

    const history = createMemoryHistory();
    history.push('/');

    render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const toDoItem = screen.getByText('ToDo 1');
    const deleteButton = screen.getByText('ÏÇ≠Ï†ú');
    expect(toDoItem).toBeInTheDocument();
    expect(deleteButton).toBeInTheDocument();

    fireEvent.click(deleteButton);
    expect(toDoItem).not.toBeInTheDocument();
    expect(deleteButton).not.toBeInTheDocument();
    expect(localStorage.getItem('ToDoList')).toBe('[]');
  });

  it('deletes ToDo from the detail page', () => {
    localStorage.setItem('ToDoList', '["ToDo 1"]');

    const history = createMemoryHistory();
    history.push('/');

    render(
      <Router history={history}>
        <App />
      </Router>,
    );

    const toDoItem = screen.getByText('ToDo 1');
    expect(toDoItem).toBeInTheDocument();
    fireEvent.click(toDoItem);
    const header = screen.getByText('Ìï† Ïùº ÏÉÅÏÑ∏');
    expect(header).toBeInTheDocument();

    const deleteButton = screen.getByText('ÏÇ≠Ï†ú');
    fireEvent.click(deleteButton);

    expect(header.textContent).toBe('Ìï† Ïùº Î™©Î°ù');
    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
    expect(toDoList.firstChild).toBeNull();
    expect(localStorage.getItem('ToDoList')).toBe('[]');
  });
});
