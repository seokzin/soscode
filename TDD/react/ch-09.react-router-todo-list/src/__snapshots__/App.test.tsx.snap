// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> goes to Add page and go back to List page 1`] = `
.c7 {
  text-align: center;
  background-color: #304FFE;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
}

.c7:hover {
  background-color: #1E40FF;
}

.c7:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c8 {
  color: #FFFFFF;
  font-size: 16px;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  padding: 10px 10px;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #1E40FF;
}

.c2 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 600;
}

.c3 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: absolute;
  left: 20px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  min-height: 100vh;
  background-color: #EEEEEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Ìï† Ïùº Ï∂îÍ∞Ä
      </div>
      <a
        class="c3"
        href="/"
      >
        ÎèåÏïÑÍ∞ÄÍ∏∞
      </a>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <input
          class="c6"
          placeholder="Ìï† ÏùºÏùÑ ÏûÖÎ†•Ìï¥ Ï£ºÏÑ∏Ïöî"
          value=""
        />
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            Ï∂îÍ∞Ä
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> goes to Detail page and go back to List page 1`] = `
.c6 {
  text-align: center;
  background-color: #FF1744;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
}

.c6:hover {
  background-color: #F01440;
}

.c6:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c7 {
  color: #FFFFFF;
  font-size: 16px;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #1E40FF;
}

.c2 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 600;
}

.c3 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: absolute;
  left: 20px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #FFFFFF;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  min-width: 350px;
  height: 350px;
  overflow-y: auto;
  border: 1px solid #BDBDBD;
  margin-bottom: 20px;
  padding: 10px;
}

.c0 {
  min-height: 100vh;
  background-color: #EEEEEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Ìï† Ïùº ÏÉÅÏÑ∏
      </div>
      <a
        class="c3"
        href="/"
      >
        ÎèåÏïÑÍ∞ÄÍ∏∞
      </a>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        ToDo 1
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          ÏÇ≠Ï†ú
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> renders component correctly 1`] = `
.c4 {
  min-width: 350px;
  height: 400px;
  overflow-y: scroll;
  border: 1px solid #BDBDBD;
  margin-bottom: 20px;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #1E40FF;
}

.c2 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 600;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 20px;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 60px;
  height: 60px;
  border-radius: 30px;
  cursor: pointer;
  position: absolute;
  bottom: -30px;
  background-color: #304FFE;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  background-color: #1E40FF;
}

.c5:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c0 {
  min-height: 100vh;
  background-color: #EEEEEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Ìï† Ïùº Î™©Î°ù
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
        data-testid="toDoList"
      />
      <a
        class="c5"
        href="/add"
      >
        +
      </a>
    </div>
  </div>
</div>
`;

exports[`<App /> shows Not Found page if the user enters the wrong URL, and go back to List page 1`] = `
.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #1E40FF;
}

.c2 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 600;
}

.c3 {
  padding: 20px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: absolute;
  left: 20px;
}

.c4 {
  font-size: 20px;
}

.c0 {
  min-height: 100vh;
  background-color: #EEEEEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        ÏóêÎü¨
      </div>
      <a
        class="c3"
        href="/"
      >
        ÎèåÏïÑÍ∞ÄÍ∏∞
      </a>
    </div>
    <div
      class="c4"
    >
      Not Found üòø
    </div>
  </div>
</div>
`;
